### PR TITLE
fixes 92907 improves test error output

### LINKED ISSUE #92907: Improve test error message : NodeProblemDetector SystemLogMonitor should generate node condition and events for corresponding error
The NodeProblemDetector test  - **SystemLogMonitor should generate node condition and events for corresponding errors** 
can produce the following error message which is misleading 

> expect event number 4, got 7:

This message outputted by verifyTotalEvents  is meant to indicate that a specific _number of occurrences of events_ is expected but an _unexpected number of events_ occurred during the test run. 

It's misleading because it can be interpreted as 

> the event number 4 occurred but instead we got event number 7

where 7 and 4 are _event IDs_ instead of _event counts_

Example https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-node-kubelet-features/1280788392149258243

### PR DESCRIPTION
**Pull request by RobertKielty**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Clarifies a misleading test error message 

**Which issue(s) this PR fixes**:

Fixes #92907

**Special notes for your reviewer**:
Please see linked issue #92907 
 
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e_node/node_problem_detector_linux.go [hunk 1]
```diff
@@ -419,7 +419,7 @@ func verifyEvents(e coreclientset.EventInterface, options metav1.ListOptions, nu
 		count += int(event.Count)
 	}
 	if count != num {
-		return fmt.Errorf("expect event number %d, got %d: %v", num, count, events.Items)
+		return fmt.Errorf("Expected %d events with reason set to %s and message set to %s\nbut %d actual events occured. Events : %v", num, reason, message, count, events.Items)
```
Comment (hasheddan) [inline test/e2e_node/node_problem_detector_linux.go:422]: nitpick: I _think_ convention is to not capitalize error message strings. Not sure if linter will actually check this or not

Comment (RobertKielty) [reply test/e2e_node/node_problem_detector_linux.go:422]: Thanks Dan,let me do some surveying on this it would be good to know and follow convention

Comment (hasheddan) [reply test/e2e_node/node_problem_detector_linux.go:422]: Looks like the linter did catch the spelling of `occurred` 😅

Comment (RobertKielty) [reply test/e2e_node/node_problem_detector_linux.go:422]: Survey complete, you are spot on, the only time fmtErrorf messages start start with a capital letter is when the message starts with a exported type and in tests error messages always start with a lowercase letter ...

https://cs.k8s.io/?q=fmt.Errorf&i=nope&files=test&repos=kubernetes/kubernetes

**Pull request merged**